Nick@~/workspace/ouplayer (2.x=)$ grep -nr ou-specific application/engines
application/engines/mediaelement/.htaccess:64:#ou-specific.
application/engines/mediaelement/build/mediaelement-and-player.js:979://ou-specific [iet-it-bugs:1416]
application/engines/mediaelement/build/mediaelement-and-player.js:985://ou-specific ends.
application/engines/mediaelement/build/mediaelement-and-player.min-OU-SP.js:44:"video":"audio");a.parentNode.insertBefore(f,a);a.style.display="none";j.htmlMediaElement=a=f}for(f=0;f<g.length;f++){/*ou-specific #1416*/var _can_play_re= c.can_play_maybe?/no/:/no|maybe/;if(a.canPlayType(g[f].type).replace(_can_play_re,"")!==''||a.canPlayType(g[f].type.replace(/mp3/,'mpeg')).replace(_can_play_re,"")!==''){/*ou-specific ends*/j.method="native";j.url=g[f].url;break}}if(j.method==="native"){if(j.url!==null)a.src=j.url;if(b.mode!=="auto_plugin")return j}}if(b.mode==="auto"||b.mode==="auto_plugin"||b.mode==="shim")for(f=0;f<g.length;f++){l=g[f].type;for(a=0;a<b.plugins.length;a++){e=
application/engines/mediaelement/build/mediaelement.js:979://ou-specific [iet-it-bugs:1416]
application/engines/mediaelement/build/mediaelement.js:985://ou-specific ends.
application/engines/mediaelement/src/js/me-shim.js:274://ou-specific [iet-it-bugs:1416]
application/engines/mediaelement/src/js/me-shim.js:280://ou-specific ends.

Nick@~/workspace/ouplayer (2.x=)$ grep -nr --exclude=*.patch ou-specific application/themes
application/themes/ouplayer_base/js/mep-feature-fullscreen.js:61://ou-specific:  NDF 2012-03-30.
application/themes/ouplayer_base/js/mep-feature-fullscreen.js:64://ou-specific ends.
application/themes/ouplayer_base/js/mep-feature-tracks.js:24://ou-specific, Forward-port, #902: https://github.com/johndyer/mediaelement/commit/c753e35e2b8aaa7a
application/themes/ouplayer_base/js/mep-feature-tracks.js:30://ou-specific ends.
application/themes/ouplayer_base/js/mep-feature-tracks.js:50://ou-specific:  NDF 2012-03-30.
application/themes/ouplayer_base/js/mep-feature-tracks.js:53://ou-specific ends.
application/themes/ouplayer_base/js/mep-oup-feature-progress.js:8://ou-specific
application/themes/ouplayer_base/js/mep-oup-feature-progress.js:13://ou-specific ends.
application/themes/ouplayer_base/js/mep-oup-feature-progress.js:17://ou-specific
application/themes/ouplayer_base/js/mep-oup-feature-progress.js:37://ou-specific
application/themes/ouplayer_base/js/mep-oup-feature-progress.js:40://ou-specific ends.
application/themes/ouplayer_base/js/mep-oup-feature-progress.js:77://ou-specific
application/themes/ouplayer_base/js/mep-oup-feature-progress.js:176://ou-specific ends.
application/themes/ouplayer_base/js/mep-oup-feature-progress.js:225://ou-specific
application/themes/ouplayer_base/js/mep-player.js:1066://ou-specific:  Ender fix, NDF 2012-04-17.
Nick@~/workspace/ouplayer (2.x=)$







Nick@~/workspace/ouplayer/application (2.x=)$ diff -u engines/mediaelement/src/js/mep-feature-fullscreen.js  themes/ouplayer_base/js/mep-feature-fullscreen.js
--- engines/mediaelement/src/js/mep-feature-fullscreen.js	2014-05-28 17:44:37.000000000 +0100
+++ themes/ouplayer_base/js/mep-feature-fullscreen.js	2014-05-28 17:44:37.000000000 +0100
@@ -3,7 +3,7 @@
 	$.extend(mejs.MepDefaults, {
 		usePluginFullScreen: true,
 		newWindowCallback: function() { return '';},
-		fullscreenText: mejs.i18n.t('Fullscreen')
+		fullscreenText: 'Fullscreen' //mejs.i18n.t('Fullscreen')
 	});

 	$.extend(MediaElementPlayer.prototype, {
@@ -58,7 +58,10 @@
 					$('<div class="mejs-button mejs-fullscreen-button">' +
 						'<button type="button" aria-controls="' + t.id + '" title="' + t.options.fullscreenText + '"></button>' +
 					'</div>')
-					.appendTo(controls);
+//ou-specific:  NDF 2012-03-30.
+					.appendTo(controls.group());
+					//.appendTo(controls);
+//ou-specific ends.

 				if (t.media.pluginType === 'native' || (!t.options.usePluginFullScreen && !mejs.MediaFeatures.isFirefox)) {

Nick@~/workspace/ouplayer/application (2.x=)$









Nick@~/workspace/ouplayer/application (2.x=)$ diff -u engines/mediaelement/src/js/mep-feature-tracks.js  themes/ouplayer_base/js/mep-feature-tracks.js
--- engines/mediaelement/src/js/mep-feature-tracks.js	2014-05-28 17:44:37.000000000 +0100
+++ themes/ouplayer_base/js/mep-feature-tracks.js	2014-05-28 17:44:37.000000000 +0100
@@ -21,6 +21,13 @@

 			var t= this, i, options = '';

+//ou-specific, Forward-port, #902: https://github.com/johndyer/mediaelement/commit/c753e35e2b8aaa7a
+                        if (t.domNode.textTracks) { // if browser will do native captions, prefer mejs captions, loop through tracks and hide
+                                for (var i = t.domNode.textTracks.length - 1; i >= 0; i--) {
+                                        t.domNode.textTracks[i].mode = "hidden";
+                                }
+                        }
+//ou-specific ends.
 			player.chapters =
 					$('<div class="mejs-chapters mejs-layer"></div>')
 						.prependTo(layers).hide();
@@ -40,7 +47,10 @@
 							'</ul>'+
 						'</div>'+
 					'</div>')
-						.appendTo(controls)
+//ou-specific:  NDF 2012-03-30.
+						.appendTo(controls.group())
+						//.appendTo(controls)
+//ou-specific ends.

 						// hover
 						.hover(function() {
@@ -164,29 +174,35 @@

 				};

+			if (track.isTranslation) {
+
+				// translate the first track
+				mejs.TrackFormatParser.translateTrackText(t.tracks[0].entries, t.tracks[0].srclang, track.srclang, t.options.googleApiKey, function(newOne) {
+
+					// store the new translation
+					track.entries = newOne;

-			$.ajax({
-				url: track.src,
-				dataType: "text",
-				success: function(d) {
-
-					// parse the loaded file
-					if (typeof d == "string" && (/<tt\s+xml/ig).exec(d)) {
-						track.entries = mejs.TrackFormatParser.dfxp.parse(d);
-					} else {
-						track.entries = mejs.TrackFormatParser.webvvt.parse(d);
-					}
-
 					after();
+				});

-					if (track.kind == 'chapters' && t.media.duration > 0) {
-						t.drawChapters(track);
+			} else {
+				$.ajax({
+					url: track.src,
+					success: function(d) {
+
+						// parse the loaded file
+						track.entries = mejs.TrackFormatParser.parse(d);
+						after();
+
+						if (track.kind == 'chapters' && t.media.duration > 0) {
+							t.drawChapters(track);
+						}
+					},
+					error: function() {
+						t.loadNextTrack();
 					}
-				},
-				error: function() {
-					t.loadNextTrack();
-				}
-			});
+				});
+			}
 		},

 		enableTrackButton: function(lang, label) {
@@ -404,106 +420,53 @@
 	Adapted from: http://www.delphiki.com/html5/playr
 	*/
 	mejs.TrackFormatParser = {
-		webvvt: {
-			// match start "chapter-" (or anythingelse)
-			pattern_identifier: /^([a-zA-z]+-)?[0-9]+$/,
-			pattern_timecode: /^([0-9]{2}:[0-9]{2}:[0-9]{2}([,.][0-9]{1,3})?) --\> ([0-9]{2}:[0-9]{2}:[0-9]{2}([,.][0-9]{3})?)(.*)$/,
-
-			parse: function(trackText) {
-				var
-					i = 0,
-					lines = mejs.TrackFormatParser.split2(trackText, /\r?\n/),
-					entries = {text:[], times:[]},
-					timecode,
-					text;
-				for(; i<lines.length; i++) {
-					// check for the line number
-					if (this.pattern_identifier.exec(lines[i])){
-						// skip to the next line where the start --> end time code should be
-						i++;
-						timecode = this.pattern_timecode.exec(lines[i]);
+		// match start "chapter-" (or anythingelse)
+		pattern_identifier: /^([a-zA-z]+-)?[0-9]+$/,
+		pattern_timecode: /^([0-9]{2}:[0-9]{2}:[0-9]{2}([,.][0-9]{1,3})?) --\> ([0-9]{2}:[0-9]{2}:[0-9]{2}([,.][0-9]{3})?)(.*)$/,

-						if (timecode && i<lines.length){
-							i++;
-							// grab all the (possibly multi-line) text that follows
-							text = lines[i];
-							i++;
-							while(lines[i] !== '' && i<lines.length){
-								text = text + '\n' + lines[i];
-								i++;
-							}
-							text = $.trim(text).replace(/(\b(https?|ftp|file):\/\/[-A-Z0-9+&@#\/%?=~_|!:,.;]*[-A-Z0-9+&@#\/%=~_|])/ig, "<a href='$1' target='_blank'>$1</a>");
-							// Text is in a different array so I can use .join
-							entries.text.push(text);
-							entries.times.push(
-							{
-								start: (mejs.Utility.convertSMPTEtoSeconds(timecode[1]) == 0) ? 0.200 : mejs.Utility.convertSMPTEtoSeconds(timecode[1]),
-								stop: mejs.Utility.convertSMPTEtoSeconds(timecode[3]),
-								settings: timecode[5]
-							});
-						}
-					}
-				}
-				return entries;
-			}
+		split2: function (text, regex) {
+			// normal version for compliant browsers
+			// see below for IE fix
+			return text.split(regex);
 		},
-		// Thanks to Justin Capella: https://github.com/johndyer/mediaelement/pull/420
-		dfxp: {
-			parse: function(trackText) {
-				trackText = $(trackText).filter("tt");
-				var
-					i = 0,
-					container = trackText.children("div").eq(0),
-					lines = container.find("p"),
-					styleNode = trackText.find("#" + container.attr("style")),
-					styles,
-					begin,
-					end,
-					text,
-					entries = {text:[], times:[]};
-
-
-				if (styleNode.length) {
-					var attributes = styleNode.removeAttr("id").get(0).attributes;
-					if (attributes.length) {
-						styles = {};
-						for (i = 0; i < attributes.length; i++) {
-							styles[attributes[i].name.split(":")[1]] = attributes[i].value;
+		parse: function(trackText) {
+			var
+				i = 0,
+				lines = this.split2(trackText, /\r?\n/),
+				entries = {text:[], times:[]},
+				timecode,
+				text;
+
+			for(; i<lines.length; i++) {
+				// check for the line number
+				if (this.pattern_identifier.exec(lines[i])){
+					// skip to the next line where the start --> end time code should be
+					i++;
+					timecode = this.pattern_timecode.exec(lines[i]);
+
+					if (timecode && i<lines.length){
+						i++;
+						// grab all the (possibly multi-line) text that follows
+						text = lines[i];
+						i++;
+						while(lines[i] !== '' && i<lines.length){
+							text = text + '\n' + lines[i];
+							i++;
 						}
-					}
-				}

-				for(i = 0; i<lines.length; i++) {
-					var style;
-					var _temp_times = {
-						start: null,
-						stop: null,
-						style: null
-					};
-					if (lines.eq(i).attr("begin")) _temp_times.start = mejs.Utility.convertSMPTEtoSeconds(lines.eq(i).attr("begin"));
-					if (!_temp_times.start && lines.eq(i-1).attr("end")) _temp_times.start = mejs.Utility.convertSMPTEtoSeconds(lines.eq(i-1).attr("end"));
-					if (lines.eq(i).attr("end")) _temp_times.stop = mejs.Utility.convertSMPTEtoSeconds(lines.eq(i).attr("end"));
-					if (!_temp_times.stop && lines.eq(i+1).attr("begin")) _temp_times.stop = mejs.Utility.convertSMPTEtoSeconds(lines.eq(i+1).attr("begin"));
-					if (styles) {
-						style = "";
-						for (var _style in styles) {
-							style += _style + ":" + styles[_style] + ";";
-						}
+						// Text is in a different array so I can use .join
+						entries.text.push(text);
+						entries.times.push(
+						{
+							start: mejs.Utility.timeCodeToSeconds(timecode[1]),
+							stop: mejs.Utility.timeCodeToSeconds(timecode[3]),
+							settings: timecode[5]
+						});
 					}
-					if (style) _temp_times.style = style;
-					if (_temp_times.start == 0) _temp_times.start = 0.200;
-					entries.times.push(_temp_times);
-					text = $.trim(lines.eq(i).html()).replace(/(\b(https?|ftp|file):\/\/[-A-Z0-9+&@#\/%?=~_|!:,.;]*[-A-Z0-9+&@#\/%=~_|])/ig, "<a href='$1' target='_blank'>$1</a>");
-					entries.text.push(text);
-					if (entries.times.start == 0) entries.times.start = 2;
 				}
-				return entries;
 			}
-		},
-		split2: function (text, regex) {
-			// normal version for compliant browsers
-			// see below for IE fix
-			return text.split(regex);
+
+			return entries;
 		}
 	};

Nick@~/workspace/ouplayer/application (2.x=)$






Nick@~/workspace/ouplayer/application (2.x=)$ diff -u engines/mediaelement/src/js/mep-player.js themes/ouplayer_base/js/mep-player.js
--- engines/mediaelement/src/js/mep-player.js	2014-05-28 17:44:37.000000000 +0100
+++ themes/ouplayer_base/js/mep-player.js	2014-05-28 17:44:37.000000000 +0100
@@ -534,18 +534,14 @@
 						});

 					} else {
-						// click controls
-						var clickElement = (t.media.pluginType == 'native') ? t.$media : $(t.media.pluginElement);
-
-						// click to play/pause
-						clickElement.click(function() {
-							if (media.paused) {
-								media.play();
-							} else {
-								media.pause();
-							}
-						});
-
+            // click to play/pause
+            t.media.addEventListener('click', function() {
+              if (t.media.paused) {
+                t.media.play();
+              } else {
+                t.media.pause();
+              }
+            });

 						// show/hide controls
 						t.container
@@ -713,22 +709,22 @@
 			if (typeof height != 'undefined')
 				t.height = height;

-			// detect 100% mode
-			if (t.height.toString().indexOf('%') > 0 || t.$node.css('max-width') === '100%') {
+      // detect 100% mode - use currentStyle for IE since css() doesn't return percentages
+      if (t.height.toString().indexOf('%') > 0 || t.$node.css('max-width') === '100%' || (t.$node[0].currentStyle && t.$node[0].currentStyle.maxWidth === '100%')) {

 				// do we have the native dimensions yet?
 				var
 					nativeWidth = t.isVideo ? ((t.media.videoWidth && t.media.videoWidth > 0) ? t.media.videoWidth : t.options.defaultVideoWidth) : t.options.defaultAudioWidth,
 					nativeHeight = t.isVideo ? ((t.media.videoHeight && t.media.videoHeight > 0) ? t.media.videoHeight : t.options.defaultVideoHeight) : t.options.defaultAudioHeight,
-					parentWidth = t.container.parent().width(),
-					newHeight = parseInt(parentWidth * nativeHeight/nativeWidth, 10);
+					parentWidth = t.container.parent().closest(':visible').width(),
+					newHeight = t.isVideo || !t.options.autosizeProgress ? parseInt(parentWidth * nativeHeight/nativeWidth, 10) : nativeHeight;

 				if (t.container.parent()[0].tagName.toLowerCase() === 'body') { // && t.container.siblings().count == 0) {
 					parentWidth = $(window).width();
 					newHeight = $(window).height();
 				}

-				if ( newHeight != 0 ) {
+				if ( newHeight != 0 && parentWidth != 0 ) {
 					// set outer container size
 					t.container
 						.width(parentWidth)
@@ -1067,6 +1063,8 @@

 	$(document).ready(function() {
 		// auto enable using JSON attribute
+//ou-specific:  Ender fix, NDF 2012-04-17.
+		if ($('.mejs-player').length > 0)
 		$('.mejs-player').mediaelementplayer();
 	});

Nick@~/workspace/ouplayer/application (2.x=)$
